quickwit-metastore/src/checkpoint.rs

```diff
@@ -123,14 +139,37 @@ pub struct Checkpoint {
     per_partition: BTreeMap<PartitionId, Position>,
 }
 
+impl Checkpoint {
+    /// Returns the number of partitions covered by the checkpoint.
+    pub fn len(&self) -> usize {
+        self.per_partition.len()
+    }
+}
+
+/// Creates a checkpoint from a `Vec` of `(u64, u64)` tuples.
```

Comment:
I'm actually only using this helper once in my tests. Happy to get rid of it if judged necessary.

Reply:
Ok, can we add a constructor for `Checkpoint` at least? Otherwise, to create a checkpoint, you have to create an empty checkpoint, then a checkpoint delta, and finally apply the delta. This is super tedious.